### PR TITLE
[ci] [release-automation] Add branch runner queue to pipeline

### PR DESCRIPTION
**Pull request by khluu**
## Changes
- Add `runner_queue_small_branch` as a runner queue option in `release-automation` pipeline

### CODE REVIEW

File: .buildkite/release-automation/config.yml [hunk 1]
```diff
@@ -13,6 +13,7 @@ runner_queues:
   macos: macos
   macos-arm64: macos-pr-arm64
   windows: windows_queue_pr
+  small_branch: runner_queue_small_branch
```
Comment (can-anyscale) [inline .buildkite/release-automation/config.yml:16]: nit: keep it as `small` for short if you want

**Review by can-anyscale — APPROVED** [review]

**Review by aslonnie — APPROVED** [review]

**Comment by aslonnie** [comment]
let me know if it is ready to merge (like if you want to use "small" instead)

**Comment by khluu** [comment]
@aslonnie  @can-anyscale I think we can just keep it as `small_branch` to specify it's a branch runner queue (it's the only one in this pipeline) and not confuse it with the rest which are PR queues

**Comment by can-anyscale** [comment]
let's do it!

**Pull request merged**